### PR TITLE
HomeWork-2

### PR DESCRIPTION
**Pull request by NastiaFasova**

### CODE REVIEW

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
      * без использования конструкций if-else</p>
      */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return (a && b || a && c || a && d || b && d || c && d || b && c)
+                && (!a && !b || !a && !c || !a && !d || !b && !d || !c && !d || !b && !c);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BooleanExpression.java]: use binary operators to simplify

Comment (NastiaFasova) [reply src/main/java/core/basesyntax/BooleanExpression.java]: Done

File: src/main/java/core/basesyntax/BooleanExpression.java [hunk 2]
```diff
@@ -1,19 +1,7 @@
...
-     *
-     * <p>Вам надо только проанализировать переданные в метод booleanExpression
-     * значения (a, b, c, d) и вернуть результат.</p>
-     *
-     * <p>Это задание необходимо решать с использованием логических операторов,
-     * без использования конструкций if-else</p>
-     */
     public boolean booleanExpression(boolean a, boolean b, boolean c, boolean d) {
-        return false;
+        return ((a ^ b) && (c ^ d) || (a ^ c) && (b ^ d));
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BooleanExpression.java]:
```suggestion
        return (a ^ b) && (c ^ d) || (a ^ c) && (b ^ d);
```

Comment (NastiaFasova) [reply src/main/java/core/basesyntax/BooleanExpression.java]: Done

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**